.circleci/config.yml

```diff
@@ -0,0 +1,380 @@
+            # resolve to anything on SauceLabs VMs ensures that such requests are always resolved
+            # through the tunnel, and resolve to the actual tunnel host machine (i.e. the CircleCI VM).
+            # More context can be found in: https://github.com/angular/angular/pull/35171.
+            setPublicVar SAUCE_LOCALHOST_ALIAS_DOMAIN "angular-ci.local"
+            setSecretVar SAUCE_ACCESS_KEY $(echo $SAUCE_ACCESS_KEY | rev)
+      - run:
+          # Sets up a local domain in the machine's host file that resolves to the local
+          # host. This domain is helpful in Saucelabs tests where devices are not able to
+          # properly resolve `localhost` or `127.0.0.1` through the sauce-connect tunnel.
+          name: Setting up alias domain for local host.
+          command: echo "127.0.0.1 $SAUCE_LOCALHOST_ALIAS_DOMAIN" | sudo tee -a /etc/hosts
+
+  start_saucelabs:
+    steps:
+      - run:
+          name: Starting Saucelabs tunnel service
+          command: ./lib/saucelabs/sauce-service.sh start-ready-wait
+
+  stop_saucelabs:
+    steps:
+      - run:
+          name: Stopping Saucelabs tunnel service
+          command: ./lib/saucelabs/sauce-service.sh stop
+
+
+  run_e2e_tests:
+    parameters:
+      specs:
+        type: string
+    steps:
+      - custom_attach_workspace
+      - init_environment
+      - init_saucelabs_environment
+      - start_saucelabs
+      - run:
+          command: yarn grunt test:circleci-protractor --specs="<< parameters.specs >>"
+          no_output_timeout: 30m
+      - stop_saucelabs
+
+  run_e2e_tests_jquery:
+    parameters:
+      specs:
+        type: string
+    steps:
+      - custom_attach_workspace
+      - init_environment
+      - init_saucelabs_environment
+      - start_saucelabs
+      - run:
+          environment:
+              USE_JQUERY: 1
+          command: yarn grunt test:circleci-protractor --specs="<< parameters.specs >>"
+          no_output_timeout: 30m
+      - stop_saucelabs
+
+# Job definitions
+# Jobs can include parameters that are passed in the workflow job invocation.
+# https://circleci.com/docs/2.0/reusing-config/#authoring-parameterized-jobs
+jobs:
+  setup:
+    executor: default-executor
+    steps:
+      - checkout
+      - init_environment
+      - run:
+          name: Running Yarn install
+          command: yarn install --frozen-lockfile --non-interactive
+          # Yarn's requests sometimes take more than 10mins to complete.
+          no_output_timeout: 45m
+      - run: yarn grunt package
+      # Persist any changes at this point to be reused by further jobs.
+      # **NOTE**: To add new content to the workspace, always persist on the same root.
+      - persist_to_workspace:
+          root: *workspace_location
+          paths:
+            - ./ng
+
+  lint:
+    executor: default-executor
+    steps:
+      - custom_attach_workspace
+      - init_environment
+      - run: yarn -s grunt ci-checks
```

Comment:
Why the `-s` here (but not elsewhere)?

Reply:
I copied it from the angular/angular project where it is used extensively. Then I forgot about it. I'll remove it because I don't think it makes much difference and is confusing if not used consistently.